packages/@sanity/presence-store/src/PresenceStore.js

```diff
@@ -0,0 +1,152 @@
+/* global window */
+
+import Observable from '@sanity/observable/minimal'
+import UUID from '@sanity/uuid'
+import deepEqual from 'deep-equal'
+
+const RESEND_REPORT_INTERVAL = 15000
+const STATE_TIMEOUT_INTERVAL = RESEND_REPORT_INTERVAL * 2
+
+function hashKeyForState(identity, session) {
+  return `${identity}__${session}`
+}
+
+export default class PresenceStore {
+  constructor(connection, channel) {
+    this.connection = connection
+    this.unsubscribeListener = this.connection.listen().subscribe(this.handleMessage).unsubscribe
+    this.myState = {}
+    this.states = {}
+    this.timestamps = {}
+    this.resendReportTimer = null
+    this.changeReportDebounceTimer = null
+    this.performPurgeTimer = setInterval(this.performPurge, 2000)
+    this.sessionId = UUID()
+    this.requestRollCall()
+
+    // List of functions that will be called with presence change notifications
+    this.subscriberCallbacks = []
+  }
+
+  // Close Presence Store, should be called when window closes
+  close() {
+    this.connection.sendBeacon({type: 'disconnect', session: this.sessionId})
+    this.unsubscribeListener()
+    clearTimeout(this.resendReportTimer)
+    clearInterval(this.performPurgeTimer)
+  }
+
+  // The observable that all subscribers will use to track state changes
+  presence = new Observable(observer => {
+    // Send initial state
+    observer.next(this.getStateReport())
+
+    // Set up subscription to track subsequent changes
+    const reporter = value => {
+      observer.next(value)
+    }
+    this.subscriberCallbacks.push(reporter)
+
+    // Create and return unsubscriber
+    const unsubscribe = () => {
+      this.subscriberCallbacks = this.subscriberCallbacks.filter(cb => cb != reporter)
+    }
+    return unsubscribe
+  })
+
+  // Call this to report this clients state
+  reportMyState(state) {
+    this.myState = state
+    this.sendMyState()
+  }
+
+  // Updates state for a client based on an incoming state message
+  updateClientState(msg) {
+    // Construct next state object
+    const state = Object.assign({
+      identity: msg.i
+    }, msg.m)
+    delete state.type
+
+    // The cache key for this state
+    const key = hashKeyForState(msg.i, msg.m.session)
+
+    // Only update state if state is actually changed
+    if (!deepEqual(state, this.states[key])) {
+      this.states[key] = state
+      this.handleRemoteChange()
+    }
+    // Update timestamp for this cache key
+    this.timestamps[key] = new Date()
```

Comment:
I'd just use `Date.now()` instead of an actual `Date` since you won't actually be using any of the date functionality